pkg/kubectl/cmd/portforward.go

```diff
@@ -131,6 +134,42 @@ func (f *defaultPortForwarder) ForwardPorts(method string, url *url.URL, opts Po
 	return fw.ForwardPorts()
 }
 
+// Translates service port to target port
+// It rewrites ports as needed if the Service uses targetPort.
+// It returns an error when a named targetPort can't find a match in the pod.
+func translateServicePortToTargetPort(ports []string, svc api.Service, pod api.Pod) ([]string, error) {
+	var translated []string
+	for _, port := range ports {
+		// port is in the form of [LOCAL PORT]:REMOTE PORT
+		parts := strings.Split(port, ":")
+		input := parts[0]
+		if len(parts) == 2 {
+			input = parts[1]
+		}
+		portnum, err := strconv.Atoi(input)
+		if err != nil {
+			return ports, err
+		}
+		containerPort, err := util.LookupContainerPortNumberByServicePort(svc, pod, int32(portnum))
+		if err != nil {
+			if containerPort <= 0 {
```

Comment:
error in all cases here... if the service didn't declare the port, I don't think it's valid to try to port-forward to it

Reply:
I think we went back to the original behavior (using gathered mapping), but it is much more clear how and why we reached the error.

Reply:
Found another corner case.  If ClusterIP: None, the current behavior would return the port as-is, even if the port is not declared by the service.  Let me fix that also.

Reply:
If `ClusterIP=None`, then no proxying occurs today, right? does it make sense to support port-forward on such a service?

Reply:
I think it makes sense to support port-forward even for ClusterIP=None.  A common use case for me is for pods with NetworkPolicy that restrict general access, so port-forward (not proxy) is a way for reaching it for testing/verification without changing the NetworkPolicy.

Reply:
ok, with the change to still require the service to declare the port, that seems reasonable